### PR TITLE
chore: add other peerdeps to correct version Gatsby 4

### PR DESCRIPTION
**Pull request by wardpeet**
### Documentation

Adding other peerdeps like gatsby-plugin-sharp to v4 constraints

### CODE REVIEW

File: patches/v4/2-gatsby-peerdeps.patch [hunk 1]
```diff
@@ -220,20 +220,24 @@ index 42131d9ae8..4a3c132ed9 100644
...
    "peerDependencies": {
      "@babel/core": "^7.12.3",
 -    "gatsby": "^3.0.0-next.0",
+-    "gatsby-plugin-sharp": "^3.0.0-next.0",
+-    "gatsby-source-filesystem": "^3.0.0-next.0",
 +    "gatsby": "^4.0.0-alpha-9689ff",
-     "gatsby-plugin-sharp": "^3.0.0-next.0",
-     "gatsby-source-filesystem": "^3.0.0-next.0",
++    "gatsby-plugin-sharp": "^4.0.0-alpha-9689ff",
++    "gatsby-source-filesystem": "^4.0.0-alpha-9689ff",
```
Comment (wardpeet) [inline patches/v4/2-gatsby-peerdeps.patch:235]: Actual change

File: patches/v4/2-gatsby-peerdeps.patch [hunk 2]
```diff
@@ -674,20 +678,22 @@ index e42d64224c..fa8c13922d 100644
...
-@@ -26,7 +26,7 @@
+@@ -26,8 +26,8 @@
      "remark"
    ],
    "peerDependencies": {
 -    "gatsby": "^3.0.0-next.0",
+-    "gatsby-remark-prismjs": "^3.0.0-next.0"
 +    "gatsby": "^4.0.0-alpha-9689ff",
-     "gatsby-remark-prismjs": "^3.0.0-next.0"
++    "gatsby-remark-prismjs": "^4.0.0-alpha-9689ff"
```
Comment (wardpeet) [inline patches/v4/2-gatsby-peerdeps.patch:691]: Actual change

File: patches/v4/2-gatsby-peerdeps.patch [hunk 3]
```diff
@@ -713,20 +719,22 @@ index f29417c056..afeb503c48 100644
...
-@@ -40,7 +40,7 @@
+@@ -40,8 +40,8 @@
    "license": "MIT",
    "main": "index.js",
    "peerDependencies": {
 -    "gatsby": "^3.0.0-next.0",
+-    "gatsby-plugin-sharp": "^3.0.0-next.0"
 +    "gatsby": "^4.0.0-alpha-9689ff",
-     "gatsby-plugin-sharp": "^3.0.0-next.0"
++    "gatsby-plugin-sharp": "^4.0.0-alpha-9689ff"
```
Comment (wardpeet) [inline patches/v4/2-gatsby-peerdeps.patch:732]: Actual change

File: patches/v4/2-gatsby-peerdeps.patch [hunk 4]
```diff
@@ -778,20 +786,26 @@ index a9ca0d279e..4276da3463 100644
...
+-    "gatsby-plugin-image": "^1.3.0-next.1",
+-    "gatsby-plugin-sharp": "^3.0.0-next.0",
+-    "sharp": "^0.26.0"
 +    "gatsby": "^4.0.0-alpha-9689ff",
-     "gatsby-plugin-image": "^1.3.0-next.1",
-     "gatsby-plugin-sharp": "^3.0.0-next.0",
-     "sharp": "^0.26.0"
++    "gatsby-plugin-image": "^2.0.0-alpha-9689ff",
++    "gatsby-plugin-sharp": "^4.0.0-alpha-9689ff",
++    "sharp": "^0.29.0"
```
Comment (wardpeet) [inline patches/v4/2-gatsby-peerdeps.patch:803]: Actual change

File: patches/v4/2-gatsby-peerdeps.patch [hunk 5]
```diff
@@ -921,20 +935,24 @@ index 86d076d653..10eb5e0e81 100644
...
    "license": "MIT",
    "peerDependencies": {
 -    "gatsby": "^3.0.0-next.0",
+-    "gatsby-plugin-sharp": "^3.0.0-next.0",
+-    "gatsby-transformer-sharp": "^3.0.0-next.0"
 +    "gatsby": "^4.0.0-alpha-9689ff",
-     "gatsby-plugin-sharp": "^3.0.0-next.0",
-     "gatsby-transformer-sharp": "^3.0.0-next.0"
++    "gatsby-plugin-sharp": "^4.0.0-alpha-9689ff",
++    "gatsby-transformer-sharp": "^4.0.0-alpha-9689ff"
```
Comment (wardpeet) [inline patches/v4/2-gatsby-peerdeps.patch:950]: Actual change

File: patches/v4/2-gatsby-peerdeps.patch [hunk 6]
```diff
@@ -999,20 +1017,22 @@ index 13860f834c..429a87f61b 100644
...
-@@ -23,7 +23,7 @@
+@@ -23,8 +23,8 @@
    ],
    "license": "MIT",
    "peerDependencies": {
 -    "gatsby": "^3.0.0-next.0",
+-    "gatsby-source-filesystem": "^3.0.0-next.0"
 +    "gatsby": "^4.0.0-alpha-9689ff",
-     "gatsby-source-filesystem": "^3.0.0-next.0"
++    "gatsby-source-filesystem": "^4.0.0-alpha-9689ff"
```
Comment (wardpeet) [inline patches/v4/2-gatsby-peerdeps.patch:1030]: Actual change

File: patches/v4/2-gatsby-peerdeps.patch [hunk 7]
```diff
@@ -1090,33 +1110,39 @@ index 6fe276496a..d1b2c2c734 100644
...
-@@ -32,7 +32,7 @@
+@@ -32,8 +32,8 @@
    ],
    "license": "MIT",
    "peerDependencies": {
 -    "gatsby": "^3.0.0-next.0",
+-    "gatsby-plugin-sharp": "^3.0.0-next.0"
 +    "gatsby": "^4.0.0-alpha-9689ff",
-     "gatsby-plugin-sharp": "^3.0.0-next.0"
++    "gatsby-plugin-sharp": "^4.0.0-alpha-9689ff"
```
Comment (wardpeet) [inline patches/v4/2-gatsby-peerdeps.patch:1123]: Actual change

File: patches/v4/2-gatsby-peerdeps.patch [hunk 8]
```diff
@@ -1090,33 +1110,39 @@ index 6fe276496a..d1b2c2c734 100644
...
    },
    "peerDependencies": {
 -    "gatsby": "^3.0.0-next.0",
+-    "gatsby-source-contentful": "^5.0.0-next.0",
+-    "gatsby-transformer-sharp": "^3.0.0-next.0"
 +    "gatsby": "^4.0.0-alpha-9689ff",
-     "gatsby-source-contentful": "^5.0.0-next.0",
-     "gatsby-transformer-sharp": "^3.0.0-next.0"
++    "gatsby-source-contentful": "^6.0.0-alpha-9689ff",
++    "gatsby-transformer-sharp": "^4.0.0-alpha-9689ff"
```
Comment (wardpeet) [inline patches/v4/2-gatsby-peerdeps.patch:1140]: Actual change

**Review by wardpeet — COMMENTED** [review]
I've added comments to actual changes to make it easier for y'all

**Review by LekoArts — APPROVED** [review]

**Pull request merged**